File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        i, result, n = 0, [], len(nums)
        
        while i < n:
            beg = end = i
            while end < n - 1 and nums[end] + 1 == nums[end + 1]: end += 1
            result.append(str(nums[beg]) + ("->" + str(nums[end])) *(beg != end))     
            i = end + 1
        
        return result

File: 0228-summary-ranges/0228-summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int i = 0;
        vector<string> result;
        int n = nums.size();

        while (i < n) {
            int beg = i;
            int end = i;

            while (end < n - 1 && nums[end] + 1 == nums[end + 1]) {
                end++;
            }

            result.push_back(to_string(nums[beg]) + (beg != end ? "->" + to_string(nums[end]) : ""));
            i = end + 1;
        }

        return result;
    }
};